### PR TITLE
Added simple commands feature

### PR DESCRIPTION
**Pull request by isavegas**
Added simple framework for commands feature. Only added one command atm (```/shrug```), for now.

Also, I fixed a small bug causing bleeding edge versions of YakYak to crash upon viewing certain conversations, due to lack of null checking a url.

### CODE REVIEW

**Comment by averissimo** [comment]
Thanks for the pull request!!

Are the commands supposed to be run as part of a longer message or just as a single command?

**Comment by averissimo** [comment]
I'm asking this as you can use the existing structure for emojis to add the /shrug command (i.e. add it to `emojishortcode.coffee`)

**Comment by isavegas** [comment]
My apologies. I was thinking more along the lines of providing the ability to use some of the more esoteric Hangouts features. For instance, Hangouts itself supports `/shrug` clientside, and I use it pretty often. As far as implementing it via emoji, I have them disabled. I dislike it when chat apps convert my smiley faces to emoji.

File: src/ui/views/input.coffee [hunk 1]
```diff
@@ -243,7 +243,12 @@ maybeFocus = ->
         el.focus() if el
 
 preparemessage = (ev) ->
-    if models.viewstate.convertEmoji
+    if models.viewstate.commandsEnabled
+        element = document.getElementById "message-input"
+        # we want to match / and ignore //
+        if element.value.substr(0,1) == "/" and element.value.substr(1,1) != "/"
+            element.value = performCommand(models, element.value)
+    else if models.viewstate.convertEmoji
```
Comment (averissimo) [inline src/ui/views/input.coffee]: breaks current functionality of emojis, as writing `:)` and enter does not convert emoji before sending.

just remove the else please :)

File: src/ui/views/menu.coffee [hunk 2]
```diff
@@ -240,6 +240,13 @@ templateView = (viewstate) ->
             enabled: viewstate.loggedin
             click: (it) -> action 'convertemoji', it.checked
         }
+        {
+            type: 'checkbox'
+            label: i18n.__('menu.view.commands:Enable Commands')
```
Comment (averissimo) [inline src/ui/views/menu.coffee]: Can you add `menu.view.commands:Enable Commands` to `src/locale/en.json.template`

It is not used for application itself, but as a template for all translations.

**Pull request closed without merging**